src/js/node/events.js

```diff
@@ -247,15 +257,17 @@ EventEmitterPrototype.listeners = function listeners(type) {
   if (!events) return [];
   var handlers = events[type];
   if (!handlers) return [];
-  return handlers.map(x => x.listener ?? x);
+  // reverse array since handlers are stored in reverse internally
+  return handlers.reverse().map(x => x.listener ?? x);
 };
 
 EventEmitterPrototype.rawListeners = function rawListeners(type) {
   var { _events } = this;
   if (!_events) return [];
   var handlers = _events[type];
   if (!handlers) return [];
-  return handlers.slice();
+  // reverse array since handlers are stored in reverse internally
+  return handlers.reverse().slice();
```

Comment:
It's worth looking at Node's micro opts in array cloning here:
https://github.com/nodejs/node/blob/main/lib/events.js#L883-L894